src/platforms/eglstream-kms/server/egl_output.cpp

```diff
@@ -316,6 +320,42 @@ uint32_t mgek::EGLOutput::crtc_id() const
     return static_cast<uint32_t>(crtc_id);
 }
 
+bool mgek::EGLOutput::queue_atomic_flip(FBHandle const& fb, void const* drm_event_userdata)
+{
+    auto const width = connector->modes[mode_index].hdisplay;
+    auto const height = connector->modes[mode_index].vdisplay;
+
+    std::unique_ptr<drmModeAtomicReq, void(*)(drmModeAtomicReqPtr)>
+        request{drmModeAtomicAlloc(), &drmModeAtomicFree};
+
+    /* Source viewport. Coordinates are 16.16 fixed point format */
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("SRC_X"), 0);
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("SRC_Y"), 0);
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("SRC_W"), width << 16);
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("SRC_H"), height << 16);
+
+    /* Destination viewport. Coordinates are *not* 16.16 */
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("CRTC_X"), 0);
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("CRTC_Y"), 0);
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("CRTC_W"), width);
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("CRTC_H"), height);
+
+    /* Set a surface for the plane */
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("CRTC_ID"), crtc_id());
+    drmModeAtomicAddProperty(request.get(), plane_id, plane_props->id_for("FB_ID"), fb);
+
+    /* We don't monitor the DRM events (yet), so have no userdata */
+    auto ret = drmModeAtomicCommit(drm_fd, request.get(), DRM_MODE_ATOMIC_NONBLOCK | DRM_MODE_PAGE_FLIP_EVENT, const_cast<void*>(drm_event_userdata));
+
+    if (ret != 0)
+    {
+        BOOST_THROW_EXCEPTION((std::system_error{-ret, std::system_category(), "Failed to commit Atomic KMS state"}));
```

Comment:
We should *at least* fall back to trying again with `ALLOW_MODESET` (that would require sharing some more state with `configure`, notably the “mode blob”).

I don't *think* we need to have a fallback to drmModeSetCrtc, as we've guaranteed we have the atomic API available

Reply:
Well... actually it seems like configure _should_ do the ALLOW_MODESET for us huh?